src/lib/components/dropdown-menu.svelte

```diff
@@ -0,0 +1,115 @@
+<script lang="ts">
+  import Icon from 'svelte-fa';
+  import { faCheck } from '@fortawesome/free-solid-svg-icons';
+
+  import { page } from '$app/stores';
+  import { goto } from '$app/navigation';
+
+  import { faCaretDown } from '@fortawesome/free-solid-svg-icons';
+  import { onMount } from 'svelte';
+  import { scale } from 'svelte/transition';
+  import IconButton from './icon-button.svelte';
+
+  import { updateQueryParameters } from '$lib/utilities/update-query-parameters';
+
+  export let value: string | undefined;
+  export let parameter: string;
+  export let options: OptionLabel[];
+
+  let show: boolean = false;
+  let menu: any = null;
+  $: _value = $page.url?.searchParams?.get(parameter) ?? undefined;
```

Comment:
You probably don't need to fall back to `undefined` since the optional chaining will do that, right?